### PR TITLE
Remove compactProof from jsigs

### PR DESCRIPTION
**Pull request by dmitrizagidulin**
This PR removes all `compactProof` usage in jsigs.

### CODE REVIEW

File: lib/ProofSet.js [hunk 1]
```diff
@@ -77,18 +69,9 @@ module.exports = class ProofSet {
...
-
-      // shallow copy document to allow removal of existing proofs
-      input = {...document};
-    }
+    // let input;
+    // shallow copy document to allow removal of existing proofs
+    const input = {...document};
 
     // save but exclude any existing proof(s)
     const proofProperty = 'proof';
```
Comment (dlongley) [inline lib/ProofSet.js:77]: There's only one proof property now, (signature being removed), so we don't need this variable.

File: lib/ProofSet.js [hunk 2]
```diff
@@ -98,37 +81,16 @@ module.exports = class ProofSet {
...
-      // add new proof
-      delete proof['@context'];
-      jsonld.addValue(document, proofProperty, proof);
-    }
+    // in-place restore any existing proofs
+    /*if(existingProofs) {
+      document[proofProperty] = existingProofs;
+    }*/
+    // add new proof
+    delete proof['@context'];
```
Comment (dlongley) [inline lib/ProofSet.js:92]: I think we can remove all of these lines (the commented code and the code deleting the context from the proof).

File: lib/ProofSet.js [hunk 3]
```diff
@@ -213,13 +165,13 @@ module.exports = class ProofSet {
 
       // get proofs from document
       const {proofSet, document: doc} = await _getProofs({
-        document, documentLoader, expansionMap, compactProof});
+        document, documentLoader, expansionMap});
```
Comment (dlongley) [inline lib/ProofSet.js:168]:
```suggestion
        document, documentLoader, expansionMap
      });
```

File: lib/ProofSet.js [hunk 4]
```diff
@@ -213,13 +165,13 @@ module.exports = class ProofSet {
...
       document = doc;
 
       // verify proofs
       const results = await _verify({
         document, suites, proofSet,
-        purpose, documentLoader, expansionMap, compactProof});
+        purpose, documentLoader, expansionMap});
```
Comment (dlongley) [inline lib/ProofSet.js:174]:
```suggestion
        purpose, documentLoader, expansionMap
      });
```

File: lib/jsonld-signatures.js [hunk 5]
```diff
@@ -38,22 +38,14 @@ const VerificationError = require('./VerificationError');
...
- *   context for internal and extension processing; this is the default
- *   behavior. To override this behavior and optimize away this step because
- *   the caller knows that the input document's JSON-LD `@context` defines
- *   the proof terms in the same way, set this flag to `false`.
  *
  * @returns {Promise<object>} Resolves with signed document.
  */
 api.sign = async function sign(document, {
-  suite, purpose, documentLoader, expansionMap, compactProof} = {}) {
+  suite, purpose, documentLoader, expansionMap} = {}) {
```
Comment (dlongley) [inline lib/jsonld-signatures.js:45]:
```suggestion
  suite, purpose, documentLoader, expansionMap
} = {}) {
```

**Review by dlongley — APPROVED** [review]
Yay, this is great. Some style/crufty code nits.

File: lib/ProofSet.js (hunk 1 above)
Comment (dmitrizagidulin) [reply lib/ProofSet.js]: Removed.

File: lib/ProofSet.js (hunk 2 above)
Comment (dmitrizagidulin) [reply lib/ProofSet.js]: Deleted.

File: lib/ProofSet.js (hunk 3 above)
Comment (dmitrizagidulin) [reply lib/ProofSet.js]: Applied, thanks.

**Comment by dmitrizagidulin** [comment]
@dlongley applied all your formatting suggestions.

**Pull request merged**